pkg/reconciler/v1alpha1/service/service.go

```diff
@@ -209,11 +219,17 @@ func (c *Reconciler) updateStatus(desired *v1alpha1.Service) (*v1alpha1.Service,
 	if reflect.DeepEqual(service.Status, desired.Status) {
 		return service, nil
 	}
+	becomesRdy := desired.Status.IsReady() && !service.Status.IsReady()
 	// Don't modify the informers copy
 	existing := service.DeepCopy()
 	existing.Status = desired.Status
 	// TODO: for CRD there's no updatestatus, so use normal update.
-	return c.ServingClientSet.ServingV1alpha1().Services(desired.Namespace).Update(existing)
+	svc, err := c.ServingClientSet.ServingV1alpha1().Services(desired.Namespace).Update(existing)
+	if err == nil && becomesRdy {
+		duration := time.Now().Sub(svc.ObjectMeta.CreationTimestamp.Time)
```

Comment:
I believe after the service is scaled down to 0 `isReady` will be false. Then when it scales back up we will reevaluate this metric as the time since original object creation.

Reply:
That makes sense. What else could I check to avoid this reevaluation?

Reply:
Actually, I just checked and the Ready condition doesn't change when the service scales to 0. So, this should be fine.